app/(dashboard)/notes/page.tsx

```diff
@@ -0,0 +1,79 @@
+import NotesTable, { NoteTableView } from '@/components/NotesTable';
+import { createClient } from '@/utils/supabase/server';
+import Link from 'next/link';
+import { redirect } from 'next/navigation';
+import { PiCaretLeft } from 'react-icons/pi';
+
+export default async function Notes({
+  searchParams,
+}: {
+  searchParams: { search?: string; tag?: string };
+}) {
+  const supabase = createClient();
+
+  const {
+    data: { user },
+  } = await supabase.auth.getUser();
+
+  if (!user) {
+    return redirect('/login');
+  }
+
+  const { search, tag } = searchParams;
+
+  if (!search && !tag) {
+    return redirect('/');
+  }
+
+  let notes: NoteTableView[] | null = [];
+
+  if (tag) {
+    const { data: notesByTags } = await supabase
+      .from('tags')
+      .select(`note_tags ( notes ( id, title, note_tags ( tags ( name ) ) ) )`)
+      .eq('name', tag);
+
+    if (notesByTags?.length) {
+      const notesByTagsMapped: NoteTableView[] = [];
+      notesByTags[0].note_tags.map((note_tag) => {
+        if (note_tag.notes) notesByTagsMapped.push(note_tag.notes);
+      });
```

Comment:
This could just be a reduce, saves pushing to an array.
```
const notesByTagsMapped = notesByTags[0].note_tags.reduce<NoteTableView[]>((mappedNotes, note) => {
    if (note_tag.notes) return [...mappedNotes, note_tag.notes];
    return mappedNotes;
}, [])
```

Or if you are not worried about looping the array twice
```
const notesByTagsMapped = notesByTags[0].note_tags.map((note_tag) => note_tag.notes).filter(Boolean);
```